File: app/src/main/java/org/baole/oned/util/markdown/InlineParser.java
package org.baole.oned.util.markdown;

import org.commonmark.parser.delimiter.DelimiterProcessor;

import java.lang.reflect.Field;
import java.util.List;
import java.util.Map;

public class InlineParser extends org.commonmark.internal.InlineParserImpl {
    public InlineParser(List<DelimiterProcessor> processorList) {
        super(processorList);

        try {
            Field f = org.commonmark.internal.InlineParserImpl.class.getDeclaredField("delimiterProcessors");
            f.setAccessible(true);
            Map<Character, DelimiterProcessor> processors = (Map<Character, DelimiterProcessor>) f.get(this);
            processors.put('*', new AsteriskDelimiterProcessor());
        } catch (Throwable e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/org/baole/oned/util/markdown/InlineParser.kt
package org.baole.oned.util.markdown

import org.commonmark.internal.InlineParserImpl
import org.commonmark.parser.delimiter.DelimiterProcessor

class InlineParser(processorList: List<DelimiterProcessor?>?) : InlineParserImpl(processorList) {
    init {
        try {
            val f = InlineParserImpl::class.java.getDeclaredField("delimiterProcessors")
            f.isAccessible = true
            val processors = f[this] as MutableMap<Char, DelimiterProcessor>
            processors['*'] = AsteriskDelimiterProcessor()
        } catch (e: Throwable) {
            e.printStackTrace()
        }
    }
}